python/lsst/obs/base/ingest.py

```diff
@@ -479,6 +530,39 @@ def run(self, files, *, pool: Optional[Pool] = None, processes: int = 1, run: Op
             if this_run not in runs:
                 self.butler.registry.registerCollection(this_run, type=CollectionType.RUN)
                 runs.add(this_run)
-            with self.butler.transaction():
-                refs.extend(self.ingestExposureDatasets(exposure, run=this_run))
+            try:
+                with self.butler.transaction():
+                    refs.extend(self.ingestExposureDatasets(exposure, run=this_run))
```

Comment:
This reminds me that if we fail here, the `exposure` dimension record does still remain inserted, and has to, because `syncDimensionData` can't be put inside a transaction.  Because `syncDimensionData` is idempotent, that's probably okay, but if we ever see a failure mode where the exposure record is actually wrong but we only notice when we get here, we should prioritize DM-24355 to get that fixed.